src/analyzer/analyzerbeats.cpp

```diff
@@ -149,41 +149,45 @@ bool AnalyzerBeats::shouldAnalyze(TrackPointer tio) const {
     // If the track already has a Beats object then we need to decide whether to
     // analyze this track or not.
     mixxx::BeatsPointer pBeats = tio->getBeats();
-    if (pBeats) {
-        QString version = pBeats->getVersion();
-        QString subVersion = pBeats->getSubVersion();
+    if (!pBeats) {
+        return true;
+    }
+    if (!mixxx::Bpm::isValidValue(pBeats->getBpm())) {
```

Comment:
This is a logical change. Bpm::isValidValue checks if the BPM is < 0. The old code checks if the BPM == 0. Do we want <= 0?

Reply:
No, that's correct. It checks for value > kValueMin = 0.0, i.e. re-analyze if bpm <= 0.